service/src/main/java/uk/nhs/adaptors/gpc/consumer/filters/SspRoutingFiler.java

```diff
@@ -0,0 +1,50 @@
+package uk.nhs.adaptors.gpc.consumer.filters;
+
+import java.util.List;
+
+import org.springframework.beans.factory.ObjectProvider;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.gateway.config.HttpClientProperties;
+import org.springframework.cloud.gateway.filter.NettyRoutingFilter;
+import org.springframework.cloud.gateway.filter.headers.HttpHeadersFilter;
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.stereotype.Component;
+import org.springframework.web.server.ServerWebExchange;
+
+import io.netty.handler.ssl.SslContext;
+import reactor.netty.http.client.HttpClient;
+import uk.nhs.adaptors.gpc.consumer.utils.PemFormatter;
+
+@Component
+public class SspRoutingFiler extends NettyRoutingFilter {
```

Comment:
Suggestions here:
Name: TlsMutualAuthRoutingFilter

We should be able to use TLS MA with or without SSP enabled. Instead of sspEnabled.equals("true") use the all-or-nothing logic from shouldBuildSslContext()